File: lib/screens/torneos_screen.dart
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';
import '../utils/colors.dart';
import '../models/ahba_models.dart';
import '../services/data_service.dart';
import 'torneo_detalle_screen.dart';

class TorneosScreen extends StatefulWidget {
  const TorneosScreen({super.key});

  @override
  State<TorneosScreen> createState() => _TorneosScreenState();
}

class _TorneosScreenState extends State<TorneosScreen> {
  String _busqueda = '';
  String _filtroRama = 'Todas'; // 'Todas', 'F', 'M'
  String _filtroCategoria = 'Todas';
  
  List<TorneoResumen> _allTorneos = [];
  List<TorneoResumen> _filteredTorneos = [];
  bool _loading = true;

  @override
  void initState() {
    super.initState();
    _loadTorneos();
  }

  Future<void> _loadTorneos() async {
    setState(() => _loading = true);
    await DataService.instance.init();
    final list = await DataService.instance.getTorneosResumen();
    if (!mounted) return;
    setState(() {
      // Filtrar para que solo aparezcan torneos de 2026 y no repetidos
      final seenNames = <String>{};
      _allTorneos = list.where((t) {
        if (t.temporada != '2026') return false;
        final key = '${t.nombre}-${t.rama}-${t.categoria}-${t.division}';
        if (seenNames.contains(key)) return false;
        seenNames.add(key);
        return true;
      }).toList();
      
      _loading = false;
      _applyFilters();
    });
  }

  void _applyFilters() {
    setState(() {
      _filteredTorneos = _allTorneos.where((t) {
        if (_filtroRama != 'Todas' && t.rama != _filtroRama) return false;
        if (_filtroCategoria != 'Todas' &&
            !t.categoria.toLowerCase().contains(_filtroCategoria.toLowerCase())) {
          return false;
        }
        if (_busqueda.isNotEmpty) {
          final q = _busqueda.toLowerCase();
          final matchNombre = t.nombre.toLowerCase().contains(q);
          final matchCat = t.categoria.toLowerCase().contains(q);
          final matchDiv = t.division.toLowerCase().contains(q);
          if (!matchNombre && !matchCat && !matchDiv) return false;
        }
        return true;
      }).toList();
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.background,
      body: Column(
        children: [
          // ========== HEADER ==========
          Container(
            padding: const EdgeInsets.fromLTRB(20, 48, 20, 20),
            decoration: const BoxDecoration(
              gradient: LinearGradient(
                colors: [AppColors.primary, Color(0xFF0F2B48)],
                begin: Alignment.topLeft,
                end: Alignment.bottomRight,
              ),
              borderRadius: BorderRadius.only(
                bottomLeft: Radius.circular(24),
                bottomRight: Radius.circular(24),
              ),
            ),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Row(
                          children: [
                            const Icon(Icons.emoji_events, color: AppColors.secondary, size: 28),
                            const SizedBox(width: 8),
                            Text(
                              'TABLA DE POSICIONES',
                              style: GoogleFonts.montserrat(
                                fontSize: 18,
                                fontWeight: FontWeight.w900,
                                color: Colors.white,
                                letterSpacing: -0.5
                              ),
                            ),
                          ],
                        ),
                        const SizedBox(height: 4),
                        const Text(
                          'Temporada Oficial 2026',
                          style: TextStyle(
                            fontSize: 12,
                            color: Colors.white70,
                            fontWeight: FontWeight.w500
                          ),
                        ),
                      ],
                    ),
                    IconButton(
                      icon: const Icon(Icons.refresh, color: Colors.white),
                      onPressed: _loadTorneos,
                    ),
                  ],
                ),
                const SizedBox(height: 16),
                
                // BARRA DE BÚSQUEDA
                TextField(
                  onChanged: (val) {
                    _busqueda = val;
                    _applyFilters();
                  },
                  style: const TextStyle(color: Colors.white),
                  decoration: InputDecoration(
                    hintText: 'Buscar torneo o división...',
                    hintStyle: const TextStyle(color: Colors.white54, fontSize: 13),
                    prefixIcon: const Icon(Icons.search, color: Colors.white70, size: 20),
                    filled: true,
                    fillColor: Colors.white.withOpacity(0.12),
                    contentPadding: const EdgeInsets.symmetric(vertical: 0),
                    border: OutlineInputBorder(
                      borderRadius: BorderRadius.circular(12),
                      borderSide: BorderSide.none,
                    ),
                  ),
                ),
              ],
            ),
          ),

          // ========== FILTROS ==========
          Container(
            padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 12),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Row(
                  children: [
                    _buildRamaChip('Damas', 'F'),
                    const SizedBox(width: 8),
                    _buildRamaChip('Caballeros', 'M'),
                  ],
                ),
                const SizedBox(height: 8),

                SingleChildScrollView(
                  scrollDirection: Axis.horizontal,
                  child: Row(
                    children: [
                      'Todas',
                      'Primera',
                      'Intermedia',
                      'Segunda',
                      'Cuarta',
                      'Quinta',
                      'Sexta',
                      'Septima',
                      'Octava',
                      'Novena',
                      '10ma',
                    ].map((cat) {
                      final isSelected = _filtroCategoria == cat;
                      return Padding(
                        padding: const EdgeInsets.only(right: 6),
                        child: FilterChip(
                          label: Text(cat, style: const TextStyle(fontSize: 11)),
                          selected: isSelected,
                          onSelected: (selected) {
                            setState(() {
                              _filtroCategoria = cat;
                              _applyFilters();
                            });
                          },
                          selectedColor: AppColors.primary,
                          backgroundColor: Colors.white,
                          labelStyle: TextStyle(
                            color: isSelected ? Colors.white : AppColors.textPrimary,
                            fontWeight: isSelected ? FontWeight.bold : FontWeight.normal,
                          ),
                          shape: RoundedRectangleBorder(
                            borderRadius: BorderRadius.circular(20),
                            side: BorderSide(
                              color: isSelected ? AppColors.primary : Colors.grey.shade300,
                            ),
                          ),
                        ),
                      );
                    }).toList(),
                  ),
                ),
              ],
            ),
          ),

          // ========== LISTA DE TORNEOS ==========
          Expanded(
            child: _loading
                ? const Center(child: CircularProgressIndicator(color: AppColors.primary))
                : _filteredTorneos.isEmpty
                    ? const Center(child: Text('No hay torneos registrados para esta selección.'))
                    : ListView.builder(
                        padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 8),
                        itemCount: _filteredTorneos.length,
                        itemBuilder: (context, index) {
                          return _buildTorneoCard(_filteredTorneos[index]);
                        },
                      ),
          ),
        ],
      ),
    );
  }

  Widget _buildRamaChip(String label, String value) {
    final isSelected = _filtroRama == value;
    return ChoiceChip(
      label: Text(label),
      selected: isSelected,
      onSelected: (selected) {
        setState(() {
          _filtroRama = selected ? value : 'Todas';
          _applyFilters();
        });
      },
      selectedColor: AppColors.secondary,
      backgroundColor: Colors.white,
      labelStyle: TextStyle(
        color: isSelected ? Colors.white : AppColors.textPrimary,
        fontWeight: FontWeight.bold,
        fontSize: 12,
      ),
    );
  }

  Widget _buildTorneoCard(TorneoResumen torneo) {
    return Card(
      margin: const EdgeInsets.only(bottom: 12),
      elevation: 0,
      color: Colors.white,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(16),
        side: BorderSide(color: Colors.grey.shade100)
      ),
      child: InkWell(
        onTap: () {
          Navigator.push(
            context,
            MaterialPageRoute(
              builder: (context) => TorneoDetalleScreen(
                torneoResumen: torneo,
                mode: TorneoDetalleMode.posiciones,
              ),
            ),
          );
        },
        borderRadius: BorderRadius.circular(16),
        child: Padding(
          padding: const EdgeInsets.all(16),
          child: Row(
            children: [
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      torneo.nombre,
                      style: GoogleFonts.montserrat(
                        fontSize: 15,
                        fontWeight: FontWeight.bold,
                        color: AppColors.textPrimary,
                      ),
                    ),
                    const SizedBox(height: 6),
                    Row(
                      children: [
                        Container(
                          padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 3),
                          decoration: BoxDecoration(
                            color: AppColors.primary.withOpacity(0.08),
                            borderRadius: BorderRadius.circular(6),
                          ),
                          child: Text(
                            torneo.categoria,
                            style: const TextStyle(
                              fontSize: 10,
                              fontWeight: FontWeight.w800,
                              color: AppColors.primary,
                            ),
                          ),
                        ),
                        if (torneo.division.isNotEmpty) ...[
                          const SizedBox(width: 8),
                          Text(
                            '•  División ${torneo.division}',
                            style: const TextStyle(
                              fontSize: 11,
                              color: Colors.grey,
                              fontWeight: FontWeight.w500
                            ),
                          ),
                        ],
                      ],
                    ),
                  ],
                ),
              ),
              const Icon(Icons.arrow_forward_ios_rounded, color: Colors.grey, size: 16),
            ],
          ),
        ),
      ),
    );
  }
}

File: app/src/main/java/com/example/hockey_app/ui/screens/torneos/TorneosScreen.kt
package com.example.hockey_app.ui.screens.torneos

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.hilt.navigation.compose.hiltViewModel
import com.example.hockey_app.data.models.TorneoResumen

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun TorneosScreen(
    onMenuClick: () -> Unit,
    onTorneoClick: (TorneoResumen) -> Unit,
    viewModel: TorneosViewModel = hiltViewModel()
) {
    val state by viewModel.state.collectAsState()
    val busqueda by viewModel.busqueda.collectAsState()
    val filtroRama by viewModel.filtroRama.collectAsState()
    val filtroCategoria by viewModel.filtroCategoria.collectAsState()

    Scaffold(
        containerColor = Color(0xFFF5F5F5)
    ) { padding ->
        Column(modifier = Modifier.padding(padding).fillMaxSize()) {
            // Header
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(bottomStart = 24.dp, bottomEnd = 24.dp))
                    .background(
                        Brush.linearGradient(
                            colors = listOf(MaterialTheme.colorScheme.primary, Color(0xFF0F2B48))
                        )
                    )
                    .padding(top = 48.dp, bottom = 20.dp, start = 20.dp, end = 20.dp)
            ) {
                Column {
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            IconButton(onClick = onMenuClick) {
                                Icon(Icons.Default.Menu, contentDescription = null, tint = Color.White)
                            }
                            Column {
                                Row(verticalAlignment = Alignment.CenterVertically) {
                                    Icon(Icons.Default.EmojiEvents, contentDescription = null, tint = MaterialTheme.colorScheme.secondary, modifier = Modifier.size(24.dp))
                                    Spacer(modifier = Modifier.width(8.dp))
                                    Text(
                                        "TABLA DE POSICIONES",
                                        color = Color.White,
                                        fontWeight = FontWeight.Black,
                                        fontSize = 16.sp
                                    )
                                }
                                Text("Temporada Oficial 2026", color = Color.White.copy(alpha = 0.7f), fontSize = 10.sp)
                            }
                        }
                        IconButton(onClick = viewModel::refresh) {
                            Icon(Icons.Default.Refresh, contentDescription = null, tint = Color.White, modifier = Modifier.size(20.dp))
                        }
                    }

                    Spacer(modifier = Modifier.height(16.dp))

                    OutlinedTextField(
                        value = busqueda,
                        onValueChange = viewModel::onSearchChange,
                        placeholder = { Text("Buscar torneo o división...", color = Color.White.copy(alpha = 0.5f), fontSize = 13.sp) },
                        modifier = Modifier.fillMaxWidth(),
                        leadingIcon = { Icon(Icons.Default.Search, contentDescription = null, tint = Color.White.copy(alpha = 0.7f), modifier = Modifier.size(20.dp)) },
                        shape = RoundedCornerShape(12.dp),
                        singleLine = true,
                        colors = OutlinedTextFieldDefaults.colors(
                            focusedContainerColor = Color.White.copy(alpha = 0.12f),
                            unfocusedContainerColor = Color.White.copy(alpha = 0.12f),
                            focusedBorderColor = Color.Transparent,
                            unfocusedBorderColor = Color.Transparent,
                            cursorColor = Color.White,
                            focusedTextColor = Color.White,
                            unfocusedTextColor = Color.White
                        )
                    )
                }
            }

            // Filters
            Column(modifier = Modifier.padding(16.dp)) {
                Row {
                    RamaChip("Damas", filtroRama == "F") { viewModel.onRamaChange(if (it) "F" else "Todas") }
                    Spacer(modifier = Modifier.width(8.dp))
                    RamaChip("Caballeros", filtroRama == "M") { viewModel.onRamaChange(if (it) "M" else "Todas") }
                }

                Spacer(modifier = Modifier.height(8.dp))

                val categorias = listOf("Todas", "Primera", "Intermedia", "Segunda", "Cuarta", "Quinta", "Sexta", "Septima", "Octava", "Novena", "10ma")
                LazyRow(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
                    items(categorias) { cat ->
                        FilterChip(
                            selected = filtroCategoria == cat,
                            onClick = { viewModel.onCategoriaChange(cat) },
                            label = { Text(cat, fontSize = 11.sp) },
                            shape = RoundedCornerShape(20.dp),
                            colors = FilterChipDefaults.filterChipColors(
                                selectedContainerColor = MaterialTheme.colorScheme.primary,
                                selectedLabelColor = Color.White
                            )
                        )
                    }
                }
            }

            // List
            Box(modifier = Modifier.weight(1f).fillMaxWidth()) {
                when (val torneosState = state) {
                    is TorneosState.Loading -> {
                        CircularProgressIndicator(modifier = Modifier.align(Alignment.Center))
                    }
                    is TorneosState.Success -> {
                        if (torneosState.torneos.isEmpty()) {
                            Text("No hay torneos registrados.", modifier = Modifier.align(Alignment.Center), color = Color.Gray)
                        } else {
                            LazyColumn(
                                contentPadding = PaddingValues(horizontal = 16.dp, vertical = 8.dp),
                                verticalArrangement = Arrangement.spacedBy(12.dp)
                            ) {
                                items(torneosState.torneos) { torneo ->
                                    TorneoCard(torneo) { onTorneoClick(torneo) }
                                }
                            }
                        }
                    }
                    is TorneosState.Error -> {
                        Text(torneosState.message, modifier = Modifier.align(Alignment.Center), color = MaterialTheme.colorScheme.error)
                    }
                }
            }
        }
    }
}

@Composable
fun RamaChip(label: String, isSelected: Boolean, onSelectedChange: (Boolean) -> Unit) {
    FilterChip(
        selected = isSelected,
        onClick = { onSelectedChange(!isSelected) },
        label = { Text(label, fontSize = 12.sp, fontWeight = FontWeight.Bold) },
        colors = FilterChipDefaults.filterChipColors(
            selectedContainerColor = MaterialTheme.colorScheme.secondary,
            selectedLabelColor = Color.White
        ),
        shape = RoundedCornerShape(20.dp)
    )
}

@Composable
fun TorneoCard(torneo: TorneoResumen, onClick: () -> Unit) {
    Card(
        modifier = Modifier.fillMaxWidth().clickable(onClick = onClick),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = Color.White),
        elevation = CardDefaults.cardElevation(0.dp)
    ) {
        Row(
            modifier = Modifier.padding(16.dp).fillMaxWidth(),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column(modifier = Modifier.weight(1f)) {
                Text(torneo.nombre, fontWeight = FontWeight.Bold, fontSize = 15.sp)
                Spacer(modifier = Modifier.height(6.dp))
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Surface(
                        color = MaterialTheme.colorScheme.primary.copy(alpha = 0.08f),
                        shape = RoundedCornerShape(6.dp)
                    ) {
                        Text(
                            torneo.categoria,
                            modifier = Modifier.padding(horizontal = 8.dp, vertical = 3.dp),
                            fontSize = 10.sp,
                            fontWeight = FontWeight.ExtraBold,
                            color = MaterialTheme.colorScheme.primary
                        )
                    }
                    if (torneo.division.isNotEmpty()) {
                        Spacer(modifier = Modifier.width(8.dp))
                        Text("• División ${torneo.division}", fontSize = 11.sp, color = Color.Gray, fontWeight = FontWeight.Medium)
                    }
                }
            }
            Icon(Icons.Default.ChevronRight, contentDescription = null, tint = Color.LightGray, modifier = Modifier.size(16.dp))
        }
    }
}
